src/hooks/core/chat/useFetchChannelHook.ts

```diff
@@ -37,29 +43,38 @@ const useFetchChannelHook = () => {
       anon_user_info_color_code: channelListInfo?.anonUserInfoColorCode
     };
 
-    channel.targetName = channelListInfo?.channelName;
-    channel.targetImage = channelListInfo?.channelImage;
+    newChannel.targetName = channelListInfo?.channelName;
+    newChannel.targetImage = channelListInfo?.channelImage;
 
-    channel.firstMessage = getFirstMessage(channel?.messages);
+    newChannel.firstMessage = getFirstMessage(channel?.messages);
 
     const isDeletedMessage = channel.firstMessage?.message_type === MESSAGE_TYPE_DELETED;
     if (isDeletedMessage) channel.firstMessage.text = DELETED_MESSAGE_TEXT;
 
-    channel.channel = {...channel};
+    newChannel.channel = {...channel};
     const channelType = channel?.type;
 
     try {
-      const channelList = ChannelList.fromChannelAPI(
-        channel,
-        type[channelType],
-        undefined,
-        anonUserInfo
-      );
-
-      await channelList.saveIfLatest(localDb);
-      refresh('channelList');
+      queue.addJob({
+        label: `saveChannelData-${channel?.id}`,
+        task: () => {
+          return new Promise((resolve) => {
+            const channelList = ChannelList.fromChannelAPI(
+              newChannel,
+              type[channelType],
+              undefined,
+              anonUserInfo
+            );
+
+            channelList.saveIfLatest(localDb).then(() => {
+              refresh('channelList');
+              resolve(true);
+            });
+          });
+        }
+      });
     } catch (e) {
-      console.log('error on helperChannelPromiseBuilder');
+      console.log('error on helperChannelPromiseBuilder', e);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made here are significant and improve the code quality. The use of a queue for saving channel data asynchronously is a good practice as it can help prevent blocking of the main thread. However, error handling within the job task could be improved. Currently, if an error occurs during the execution of the job task, it will not be caught and handled appropriately. Consider adding error handling inside the task function.
```diff
      queue.addJob({
        label: `saveChannelData-${channel?.id}`,
        task: () => {
          return new Promise((resolve, reject) => {
            const channelList = ChannelList.fromChannelAPI(
              newChannel,
              type[channelType],
              undefined,
              anonUserInfo
            );

            channelList.saveIfLatest(localDb).then(() => {
              refresh('channelList');
              resolve(true);
            }).catch(e => {
              console.error(`Error saving channel list: ${e}`);
              reject(e);
            });
          });
        }
      });
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->